Msg_brd - Final for Sprint 2/posts.js

```diff
@@ -0,0 +1,121 @@
+module.exports = function () {
+
+    var express = require('express');
+    var router = express.Router();
+
+    // Retrieve all posts
+    function getPosts(res, mysql, context, complete) {
+        console.log('Retrieving all posts');
+        mysql.pool.query("SELECT p.post_id, p.post_date, u.user_name, p.title, p.content FROM posts p INNER JOIN users u ON u.user_id = p.user_id ORDER BY p.post_date ASC", function (error, results, fields) {
+            if (error) {
+                res.write(JSON.stringify(error));
+                res.end();
+            }
+            context.posts = results;
+            complete();
+        });
+    }
+
+    // Retrieve a single post
+    function getSinglePost(res, mysql, context, id, complete) {
+        console.log('Retrieving a single post');
+        var sql = "SELECT p.post_id, p.post_date, u.user_name, p.title, p.content FROM posts p INNER JOIN users u ON u.user_id = p.user_id WHERE post_id = ?";
+        var inserts = [id];
+        mysql.pool.query(sql, inserts, function (error, results, fields) {
+            if (error) {
+                res.write(JSON.stringify(error));
+                res.end();

```

Comment:
"return" here, so the rest of the function doesn't proceed

Reply:
return added